archiver/direct.py

```diff
@@ -43,58 +43,117 @@ def archive_project(self, project_uuid: str) -> HcaSubmission:
         self.__archive(hca_submission)
         return hca_submission
 
-    def archive_submission(self, submission_uuid: str) -> dict:
+    def archive_submission(self, submission_uuid: str, archive_job: dict, ingest_api):
         hca_submission = self.__loader.get_submission(submission_uuid=submission_uuid)
-        archives_response = self.__archive(hca_submission, submission_uuid)
-        return archives_response
+        self.__archive(hca_submission, ingest_api, archive_job, submission_uuid)
 
-    def __archive(self, submission: HcaSubmission, sub_uuid=None):
-        archives_responses = {}
+    def __archive(self, submission: HcaSubmission, ingest_api, archive_job, sub_uuid=None):
         biosamples_responses = {}
         biostudies_responses = {}
-        ena_responses = {}
+        ena_study_response = {}
+        hca_assays_responses = {}
         if self.__biosamples_submitter:
             biosamples_responses = self.__biosamples_submitter.send_all_samples(submission)
-            archives_responses['samples'] = biosamples_responses
             if self.__has_response_errors(biosamples_responses.get('entities')):
-                return archives_responses
+                self.__handle_error_response(ingest_api, archive_job, "samples", biosamples_responses)
+                return
+            else:
+                self.__patch_archive_job_with_archive_result(ingest_api, archive_job, "samples", biosamples_responses)
 
         if self.__biostudies_submitter:
             biostudies_responses = self.__biostudies_submitter.send_all_projects(submission)
-            archives_responses['project'] = biostudies_responses
             if self.__has_response_errors([biostudies_responses]):
-                return archives_responses
+                self.__handle_error_response(ingest_api, archive_job, "project", biostudies_responses)
+                return
+            else:
```

Comment:
nitpick:
We probably don't need the `else` block in these checks as there is a `return` in the `if` block

Reply:
that is true. I did that way because it felt cleaner, but I might be wrong.
I just removed them and pushed it.